presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/ThriftIndexHandle.java

```diff
@@ -37,6 +50,7 @@ public ThriftIndexHandle(
     {
         this.schemaTableName = requireNonNull(schemaTableName, "schemaTableName is null");
         this.tupleDomain = requireNonNull(tupleDomain, "tupleDomain is null");
+        this.session = Optional.empty();
```

Comment:
Is this only used on workers?

Reply:
This constructor is called when deserializing the handle from JSON, which should only occur on workers.

Reply:
Yes, i noticed it's `@JsonCreator`. My question was -- is it used _only_ implicitly, on workers?

Reply:
Why not do something like this to make it obvious:

```
public XHandle(..., ..., ConnectorSession session) {
  delegate to private constructor
}

@JsonCreator
public jsonCreator(...) {
  delegate to constructor
}

private XHandle(..., ..., Optional<ConnectorSession> session) {
  ...
}
```

Reply:
I believe this is only called on workers. I will change this to perform delegation (since I agree that in general it's best to have a single constructor that sets fields)

Reply:
However, I don't see how constructor delegation relates to the question of when the object is deserialized from JSON.